pkg/util/pki/csr.go

```diff
@@ -221,6 +271,34 @@ func SignCertificate(template *x509.Certificate, issuerCert *x509.Certificate, p
 	return pemBytes.Bytes(), cert, err
 }
 
+func SignCSRTemplate(caCerts []*x509.Certificate, caKey crypto.Signer, template *x509.Certificate) (*issuer.IssueResponse, error) {
+	caCert := caCerts[0]
+
+	certPem, _, err := SignCertificate(template, caCert, template.PublicKey, caKey)
+	if err != nil {
+		return nil, err
+
+	}
+
+	chainPem, err := EncodeX509Chain(caCerts)
+	if err != nil {
+		return nil, err
+	}
+
+	certPem = append(certPem, chainPem...)
+
+	// encode the CA certificate to be bundled in the output
+	caPem, err := EncodeX509(caCerts[0])
```

Comment:
We've typically *not* had this as a shared piece of code as different CAs return the CA certificate differently (and some don't return it at all). Happy to leave this as-is for now, but I think this will need some changes when we implement another issuer type...

Reply:
:+1: